src/data/set/basic.lean

```diff
@@ -338,11 +338,13 @@ lemma nonempty.left (h : (s ∩ t).nonempty) : s.nonempty := h.imp $ λ _, and.l
 
 lemma nonempty.right (h : (s ∩ t).nonempty) : t.nonempty := h.imp $ λ _, and.right
 
-lemma nonempty_inter_iff_exists_right : (s ∩ t).nonempty ↔ ∃ x : t, ↑x ∈ s :=
-⟨λ ⟨x, xs, xt⟩, ⟨⟨x, xt⟩, xs⟩, λ ⟨⟨x, xt⟩, xs⟩, ⟨x, xs, xt⟩⟩
+@[simp] lemma inter_nonempty : (s ∩ t).nonempty ↔ ∃ x, x ∈ s ∧ x ∈ t := iff.rfl
```

Comment:
This new lemma is great, but I'm not sure whether the other two lemmas are worth changing; while I agree that the subtype lemmas are sort of weird, my impression is that the entire purpose of those lemmas is to support the subtype view; because the proposition view is better expressed using `inter_nonempty`.

Reply:
Right - but the other lemmas seem to use the proposition view, unless I'm mistaken? These inter ones feel like an odd exception in that sense and they're barely used in any case.

Reply:
@eric-wieser, I still didn't understand why the rest of these sorts of lemmas don't use the subtype view, if they are meant to support it.

Reply:
It simplifies to two `nonempty` statements, it doesn't unfold the definition of `set.nonempty`.

Reply:
Ah, I see - that is true.

Reply:
I've now removed this attribute.